splice_machine/src/test/java/com/splicemachine/db/impl/sql/compile/NullPredicateIT.java

```diff
@@ -190,6 +190,34 @@ public void testNullInInList_2() throws Exception {
         }
     }
 
+    @Test
+    public void testNotWithNullInInList_1() throws Exception {
+        String query = format("select count(*) from T --SPLICE-PROPERTIES useSpark=%s\n" +
+                                      "where a not in (cast(null as int), 5)", useSpark);
+
+        String expected = "1 |\n" +
+                "----\n" +
+                " 1 |";
```

Comment:
1 not in (null) is unknown meaning should not match.  I think the count should be zero here.

Reply:
Our representation of predicates as (true and pred) in the AST could be wrong.  Maybe we need to change our representation or avoid pruning some of the predicates.

Reply:
nice catch @msirek, you're right, the problem is that we're pruning this NULL expression too early from the in-list elements resulting with a semantically different NOT expression, i.e.
`not in(null, x)` evaluates to `not in(x)` instead of `unknown` which evaluates to `false` in join predicate.

as discussed with @ascend1, we special-case this by checking the parent of the in-list, if it is a NOT predicate and it contains a NULL element we simply return the NULL element in order to indirectly trigger the transformation of `not null` to `null`.